examples/custom_ksf.rs

```diff
@@ -0,0 +1,286 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+//
+// This source code is licensed under both the MIT license found in the
+// LICENSE-MIT file in the root directory of this source tree and the Apache
+// License, Version 2.0 found in the LICENSE-APACHE file in the root directory
+// of this source tree.
+
+//! Demonstrates a simple client-server password-based login protocol using
+//! OPAQUE, over a command-line interface
+//!
+//! This specific example shows how to use a custom KSF (Key Stretching
+//! Function). `scrypt` is used for this example, but any KSF can be used.
+
+use std::collections::HashMap;
+use std::process::exit;
+
+use generic_array::GenericArray;
+use opaque_ke::ciphersuite::CipherSuite;
+use opaque_ke::rand::rngs::OsRng;
+use opaque_ke::{
+    ClientLogin, ClientLoginFinishParameters, ClientRegistration,
+    ClientRegistrationFinishParameters, CredentialFinalization, CredentialRequest,
+    CredentialResponse, RegistrationRequest, RegistrationResponse, RegistrationUpload, ServerLogin,
+    ServerLoginStartParameters, ServerRegistration, ServerRegistrationLen, ServerSetup,
+};
+use rustyline::error::ReadlineError;
+use rustyline::Editor;
+
+// We can define a structure here to hold the parameters for the KSF.
+#[derive(Default)]
+struct CustomKsf {
+    params: scrypt::Params,
+    salt: Vec<u8>,
```

Comment:
As far as I understand, adding a salt doesn't make any sense with OPAQUE, it adds no cryptographic value. I think having custom parameters is all the customization one needs here.

@kevinlewi correct me if I'm wrong please.

Reply:
Agreed, the client key used by the server in the OPRF acts as the "salt"/entropy and is basically the way OPAQUE keeps the "salt" secret. I'll remove it!